src/cata_variant.h

```diff
@@ -67,11 +85,85 @@ constexpr cata_variant_type type_for()
     return type_for_impl<SimpleT>( std::make_index_sequence<num_types> {} );
 }
 
+// Inherit from this struct to easily implement convert specializations for any
+// string type
+template<typename T>
+struct convert_string {
+    using type = T;
+    static_assert( std::is_same<T, std::string>::value,
```

Comment:
If you only allow `std::string` as value for `T`, why make it a class template anyway? Just remove the template parameter and make it an actual class.

Reply:
Because the types being passed as template parameters are currently typedefs of `std::string`, and if they ever change to be some other type I wanted the error message to be obvious.